src/Tools/Source/RunTests/TestHistoryManager.cs

```diff
@@ -0,0 +1,213 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.TeamFoundation.Build.WebApi;
+using Microsoft.TeamFoundation.TestManagement.WebApi;
+using Microsoft.VisualStudio.Services.TestResults.WebApi;
+using Microsoft.VisualStudio.Services.WebApi;
+
+namespace RunTests;
+internal class TestHistoryManager
+{
+    /// <summary>
+    /// Azure devops limits the number of tests returned per request to 10000.
+    /// </summary>
+    private const int MaxTestsReturnedPerRequest = 10000;
```

Comment:
```suggestion
    private const int MaxTestsReturnedPerRequest = 10_000;
```

Reply:
TIL!

Reply:
Yeah it's a really subtle but awesome feature we added way back in C# 7.0 I think.